flake.nix

```diff
@@ -34,6 +36,12 @@
           });
     };
 
+    overlays.default = final: prev: {
+      inherit (prev.callPackage ./pkgs/nix-tools { }) darwin-rebuild darwin-option;
+
+      darwin-uninstaller = prev.callPackage ./pkgs/darwin-uninstaller { nix-darwin = self; };
```

Comment:
The uninstaller should be updated to also remove the `run` line from `synthetic.conf` and poke APFS appropriately.

It also completely removed `/etc/nix/nix.conf`, which is not great. It should restore the pre-nix-darwin backup file, if present.

It also printed an `rm` error when I ran it on my test VM:

```console
…
removing service org.nixos.activate-system
removing service org.nixos.nix-daemon
configuring networking...
rm: cannot remove '/Library/LaunchDaemons/org.nixos.nix-daemon.plist': No such file or directory
grep: /etc/bashrc: No such file or directory
Found no nix-daemon.sh reference in /etc/bashrc
add this snippet back to /etc/bashrc:
…
```

Maybe that's expected though.

Reply:
I think the `darwin-uninstaller` needs extra work and my preference would be to focus on that in another PR